### ISSUE TITLE
MD046 doesn't seem to work right

### ISSUE THREAD

**Issue by docwhat** [markdownlint/markdownlint#233]
**`test.md`**:

```md
-   List Item

    ```sh
    echo
    ```
```

With the defaults (supposedly 'fenced') this produces an MD046 error.

However, if you manually set the style to any of the choices ('fenced', 'consistent', 'indented') the above does not produce an error!!

```rb
rule 'MD046', style: 'indented' # 'fenced' or 'consistent'
```

**Comment by docwhat** [comment]
related: #221

**Comment by jaymzh** [comment]
I believe the attached PR solves this.
